marconi-chain-index/src/Marconi/ChainIndex/Experimental/Indexers/EpochState.hs

```diff
@@ -0,0 +1,595 @@
+            (O.encodeDisk codecConfig)
+          . extLedgerState
+
+      deserialiseLedgerState (EpochMetadata Nothing _) = const (Right Nothing)
+      deserialiseLedgerState (EpochMetadata (Just blockNo') _) =
+        bimap
+          (Text.pack . show)
+          (Just . flip EpochState blockNo' . snd)
+          . CBOR.deserialiseFromBytes
+            ( O.decodeExtLedgerState
+                (O.decodeDisk codecConfig)
+                (O.decodeDisk codecConfig)
+                (O.decodeDisk codecConfig)
+            )
+      blockNoAsText = maybe "" (Text.pack . show . (\(C.BlockNo b) -> b) . blockNo)
+      metadataAsText (Core.Timed C.ChainPointAtGenesis evt) = [blockNoAsText evt]
+      metadataAsText (Core.Timed chainPoint evt) =
+        let chainPointTexts = case chainPoint of
+              C.ChainPoint (C.SlotNo slotNo) blockHeaderHash ->
+                [Text.pack $ show slotNo, C.serialiseToRawBytesHexText blockHeaderHash]
+         in blockNoAsText evt : chainPointTexts
+   in Core.mkFileIndexer
+        path
+        False
+        "epochState"
+        "cbor"
+        serialiseLedgerState
+        deserialiseLedgerState
+        metadataAsText
+        deserialiseMetadata
+        metadataChainpoint
+
+buildBlockIndexer
+  :: (MonadIO m, MonadError Core.IndexerError m)
+  => O.CodecConfig (O.HardForkBlock (O.CardanoEras O.StandardCrypto))
+  -> FilePath
+  -> m (Core.FileIndexer EpochMetadata (C.BlockInMode C.CardanoMode))
+buildBlockIndexer codecConfig path =
+  let serialiseBlock =
+        CBOR.toLazyByteString . O.encodeDisk codecConfig . C.toConsensusBlock
+      deserialiseBlock (EpochMetadata Nothing _) = const (Right Nothing)
+      deserialiseBlock _ =
+        bimap
+          (Text.pack . show)
+          (Just . C.fromConsensusBlock C.CardanoMode)
+          . fmap (\(bs', decode) -> decode bs')
+          . CBOR.deserialiseFromBytes (O.decodeDisk codecConfig)
+      blockNoAsText = maybe "" (Text.pack . show . (\(C.BlockNo b) -> b) . getBlockNo)
+      metadataAsText (Core.Timed C.ChainPointAtGenesis evt) = [blockNoAsText evt]
+      metadataAsText (Core.Timed chainPoint evt) =
+        let chainPointTexts = case chainPoint of
+              C.ChainPoint (C.SlotNo slotNo) blockHeaderHash ->
+                [Text.pack $ show slotNo, C.serialiseToRawBytesHexText blockHeaderHash]
+         in blockNoAsText evt : chainPointTexts
+   in Core.mkFileIndexer
+        path
+        True
+        "block"
+        "cbor"
+        serialiseBlock
+        deserialiseBlock
+        metadataAsText
+        deserialiseMetadata
+        metadataChainpoint
+
+-- TODO placeholder, to implement
+buildEpochSDDIndexer
+  :: (MonadIO m, MonadError Core.IndexerError m)
+  => FilePath
+  -> m (Core.SQLiteIndexer (NonEmpty EpochSDD))
+buildEpochSDDIndexer path =
+  Core.mkSingleInsertSqliteIndexer
+    path
+    (Lens.view Core.point)
+    Sync.syncTableCreation
+    [sql|INSERT INTO sync (slotNo, blockHeaderHash) VALUES (?, ?)|]
```

Comment:
Use `Sync.syncInsertPlan`?

Reply:
Probably not an issue if we plan to change it?

Reply:
What do we plan to change?

Reply:
not using `mkSingleInsertSqliteIndexer` as we must handle the event and the point. That's another reason why I couldn't use syncInsertPlan without coercion: it's a newtype on top of `Query`.